source/module_cell/read_pp.cpp

```diff
@@ -487,3 +487,181 @@ void Pseudopot_upf::setqfnew(const int& nqf,
         rho[ir] *= pow(r[ir], l + n);
     }
 }
+
+void Pseudopot_upf::complete_default(Atom_pseudo& pp)
+{
+    ModuleBase::TITLE("Pseudopot_upf", "complete_default");
+
+    // call subroutines
+    this->complete_default_h(pp);
+	this->complete_default_atom(pp);
+	this->complete_default_vl(pp);
+
+	if (pp.nbeta == 0) {
+		return;
+	}
+	
+	if (pp.lll == nullptr)
+	{
+		pp.lll = new int[pp.nbeta];
```

Comment:
Is it possible to use std::vector to replace new and delete?

Reply:
Yes, if needed, I will replace all `new`/`delete` in `pseudo` class with `vector` in next pr.